### PR TITLE
fix(es/transforms/jsx): collapses multiple whitespace charactars

### LINKED ISSUE #2208: swc collapses multiple whitespace characters in JSX expressions.
**Describe the bug**
swc collapses two or more spaces into one space in JSX expressions. This causes unexpected formatting when using `white-space: pre`.

**Input code**
```tsx
const Component = () => {
  return <pre>|    |    |</pre>;
}
```

**Config**

```json
{
  "jsc": {
    "parser": {
      "syntax": "ecmascript",
      "jsx": true
    }
  }
}
```

**Expected behavior**

Whitespace is preserved. Compare output from swc:
```tsx
var Component = function() {
    return(/*#__PURE__*/ React.createElement("pre", null, "| | |"));
};
```

to output from babel:
```tsx
var Component = function Component() {
  return /*#__PURE__*/React.createElement("pre", null, "|    |    |");
};
```

**Version**
1.2.84

**Additional context**
This case for this is ASCII art rendered in React. The current workaround is wrapping every expression in a template string to preserve spaces.

### PR DESCRIPTION
**Pull request by RiESAEX**
attempt to fix #2208

### CODE REVIEW

File: ecmascript/transforms/react/src/jsx/mod.rs [hunk 1]
```diff
@@ -1130,42 +1130,28 @@ fn to_prop_name(n: JSXAttrName) -> PropName {
...
-    for (last, s) in s
-        .split(|c: char| c != '\u{a0}' && c.is_ascii_whitespace())
-        .filter(|s| !s.is_empty())
+    static SPACE_START: Lazy<Regex> = Lazy::new(|| Regex::new("^[ ]+").unwrap());
+    static SPACE_END: Lazy<Regex> = Lazy::new(|| Regex::new("[ ]+$").unwrap());
+    t.lines()
+        .map(|s| s.replace('\t', " "))
+        .enumerate()
+        .filter(|(_, v)| v.len() != 0)
         .identify_last()
```
Comment (kdy1) [inline ecmascript/transforms/react/src/jsx/mod.rs:1139]: This seems like an inefficient code, as it allocates lots of `String`s.
I think using a buffer is preferrable.

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thanks!

File: ecmascript/transforms/react/src/jsx/mod.rs [hunk 2]
```diff
@@ -1130,41 +1130,34 @@ fn to_prop_name(n: JSXAttrName) -> PropName {
...
-        .split(|c: char| c != '\u{a0}' && c.is_ascii_whitespace())
-        .filter(|s| !s.is_empty())
-        .identify_last()
-    {
-        buf.push_str(s);
-        if !last {
-            buf.push(' ');
+    static SPACE_START: Lazy<Regex> = Lazy::new(|| Regex::new("^[ ]+").unwrap());
+    static SPACE_END: Lazy<Regex> = Lazy::new(|| Regex::new("[ ]+$").unwrap());
+    let mut buf = String::from("");
```
Comment (kdy1) [inline ecmascript/transforms/react/src/jsx/mod.rs:1135]: Can you this change to `String::new()`?

**Review by kdy1 — CHANGES_REQUESTED** [review]
Mostly LGTM.
Thnaks!

File: ecmascript/transforms/react/src/jsx/mod.rs (hunk 2 above)
Comment (RiESAEX) [reply ecmascript/transforms/react/src/jsx/mod.rs:1135]: of course.
btw, I looked into some failed test case, noticing their output is different with the output which I got with tsc playground, should i edit them? most of them added in #2456

Comment (kdy1) [reply ecmascript/transforms/react/src/jsx/mod.rs:1135]: Yes, those tests exist to show difference of output of lots of inputs.
You can run `UPDATE=1 cargo test` from the repository root to update them.
Note that you may need to run it multiple time, until it exists normally.

**Review by kdy1 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**